database/scrapers/event_calendar.ts

```diff
@@ -0,0 +1,230 @@
+      nestedDate.length !== 13
+    ) {
+      return null;
+    }
+    const dateTime = DateTime.fromISO(nestedDate.substring(5));
+    if (dateTime.isValid) {
+      return dateTime;
+    }
+    return null;
+  }
+
+  private static assertString(value: ICSValue | undefined): string | null {
+    if (value === undefined || typeof value !== "string") {
+      return null;
+    }
+    return value;
+  }
+
+  private parseEvent(eventBlock: ICSRoot): GoogleCalendarEventDto | undefined {
+    const startTime = CalendarParser.assertValidDateTime(
+      eventBlock[FIELD_LABELS.startDate],
+    );
+    const endTime = CalendarParser.assertValidDateTime(
+      eventBlock[FIELD_LABELS.endDate],
+    );
+    if (startTime === null || endTime === null) {
+      this.task.update(
+        `Invalid calendar event ${this.events.length + 1}, missing date labels. Skipping...`,
+      );
+      return undefined;
+    }
+    const googleCalId = CalendarParser.assertString(
+      eventBlock[FIELD_LABELS.googleCalId],
+    );
+    if (googleCalId === null) {
+      this.task.update(
+        `Invalid calendar event ${this.events.length + 1}, missing googleCalId. Skipping...`,
+      );
+      return undefined;
+    }
+    const name = CalendarParser.assertString(eventBlock[FIELD_LABELS.name]);
+    if (name === null) {
+      this.task.update(
+        `Invalid calendar event ${this.events.length + 1}, missing name. Skipping...`,
+      );
+      return undefined;
+    }
+    const description = CalendarParser.assertString(
+      eventBlock[FIELD_LABELS.desc],
+    );
+    const location = CalendarParser.assertString(
+      eventBlock[FIELD_LABELS.location],
+    );
+    return {
+      googleCalId,
+      startTime,
+      endTime,
+      name,
+      location,
+      description,
+    };
+  }
+}
+
+export default class EventCalendarUpdater extends BaseScraperModule {
+  static name = "Event calendar";
+  static description =
+    "Update and push to the database events from the Student Council Google calendar as Google events (non-editable, removable)";
+  static taskTitle =
+    "Update event calendar with latest events from Student Council Google calendar";
+
+  private async fetchGoogleEvents(): Promise<string> {
+    const response = await this.fetchAndCheckStatus(
+      `https://calendar.google.com/calendar/ical/${CALENDAR_ID}/public/basic.ics`,
+      "Event calendar ICS file",
+    );
+    return response.text();
+  }
+
+  private async getEventsInTimeframe(
+    task: TaskHandle,
+  ): Promise<GoogleCalendarEventDto[]> {
+    const calendarString = await this.fetchGoogleEvents();
+    const parser: CalendarParser = new CalendarParser(calendarString, task);
+    const bound = DateTime.now().minus({ days: FETCH_LIMIT_DAYS });
+    const events: GoogleCalendarEventDto[] = [];
+    let event = parser.getLastEvent();
+    while (event !== null && event.startTime >= bound) {
+      events.push(event);
+      event = parser.getLastEvent();
+    }
```

Comment:
I get the optimization you tried to make, but from what I can see, the events in the .ics file aren't actually sorted by start/end time - it looks more like a sort by creation time.
and you know - parseICS just parses the whole thing anyway, you might as well replace this quite ugly loop with a .map.filter chain.